src/annotator/anchoring/range-js.js

```diff
@@ -32,3 +34,42 @@ export function nodeFromXPath(xpath, root) {
   });
   return node;
 }
+
+/**
+ * Finds an element node using an XPath relative to the document root.
+ *
+ * Example:
+ *   node = nodeFromXPath('/html/body/div/p[2]')
+ */
+export function nodeFromXPath(xpath, root = document) {
+  /**
+   * Attempt to evaluate a provided XPath. If the evaluation fails, then fall back to a
+   * manual evaluation using `nodeFromXPathFallback` that can evaluate very simple XPaths such
+   * as those generated by `xpathFromNode`
+   */
+  function evaluateXPath(xp, root, nsResolver = null) {
```

Comment:
I moved this back inside to group it since we're not testing for XML at all now and it makes no sense outside of the nodeFromXPath context anyway.